Component/BowArrowOnly.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Collections;
+using GameModeLoader.Data;
+using GameModeLoader.Utils;
+using ThunderRoad;
+using Wully.Utils;
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GameModeLoader.Component
+{
+	public class BowArrowOnly : LevelModuleOptional
+	{
+		private List<string> allowedItemIds = new List<string> { "Arrow", "Bow", "Quiver" };
+
+		private List<ItemData.Type> allowedItemTypes = new List<ItemData.Type> { ItemData.Type.Quiver, ItemData.Type.Prop };
+		public override IEnumerator OnLoadCoroutine()
+		{
+			//You must always call the following, so the IDs are setup for this LevelModuleOptional
+			SetId();
+
+			//Use IsEnabled around functionality so it only runs if this LevelModuleOptional is enabled
+			if (IsEnabled())
+			{
+				Debug.Log($"This levelModuleOption is enabled - {id}");
+				Utilities.BowArrowOnlyItem();
+				EventManager.onPossess += EventManager_onPossess;
+                EventManager.onUnpossess += EventManager_onUnpossess;
+			}
+			yield break;
+		}
+
+        private void EventManager_onUnpossess(Creature creature, EventTime eventTime)
+        {
+			creature.handLeft.OnGrabEvent -= HandLeft_OnGrabEvent;
+			creature.handRight.OnGrabEvent -= HandRight_OnGrabEvent;
+		}
+
+        private void EventManager_onPossess(Creature creature, EventTime eventTime)
+        {
+			if (eventTime == EventTime.OnStart)
+			{
+				Utilities.BowArrowOnlyUIItemSpawner();
+				creature.handLeft.OnGrabEvent += HandLeft_OnGrabEvent;
+				creature.handRight.OnGrabEvent += HandRight_OnGrabEvent;
+				return;
+			}
+			if (eventTime == EventTime.OnEnd)
+			{
+				creature.handLeft.caster.allowCasting = false;
+				creature.handLeft.caster.allowSpellWheel = false;
+				// This works for the no TK, it's still active but unusable
+				SpellTelekinesis.showHighlighter = false;
+				creature.handLeft.caster.telekinesis.maxCatchDistance = 0.0f;
+				creature.handLeft.caster.telekinesis.radius = 0.0f;
+				creature.handLeft.caster.telekinesis.maxAngle = 0.0f;
+				
+				creature.handRight.caster.allowCasting = false;
+				creature.handRight.caster.allowSpellWheel = false;
+				creature.handRight.caster.telekinesis.maxCatchDistance = 0.0f;
+				creature.handRight.caster.telekinesis.radius = 0.0f;
+				creature.handRight.caster.telekinesis.maxAngle = 0.0f;
+				return;
+			}
+		}
+		private void HandRight_OnGrabEvent(Side side, Handle handle, float axisPosition, HandleOrientation orientation, EventTime eventTime)
+		{
+			UngrabUnallowedItems(handle, eventTime);
+		}
+
+		private void HandLeft_OnGrabEvent(Side side, Handle handle, float axisPosition, HandleOrientation orientation, EventTime eventTime)
+		{
+			UngrabUnallowedItems(handle, eventTime);
+		}
+
+		public override void Update()
```

Comment:
you can remove this method if you arent using it, otherwise B&S will still call it every frame, which does have a performance hit